### ISSUE TITLE
Identity Requests (value attribute)

### ISSUE THREAD

**Issue by ryanbrainard** [heroku/react-refetch#33]
Add a `value` attribute to requests (aka mappings) the return the `value` in lieu of actually fetching data from `url`. Requests with `value` are not allowed to also have a `value`.

This will support composable transformations, like this (from #21):

```jsx
connect((props) => ({
  data: {
    url: "/api/data",
    then: (rawValue) => ({ 
      value: rawValue.real_data 
    })
  }
}))(Component)
```

and will also provide a way to create static `PromiseState`s, like this (from #19):

```jsx
connect((props) => ({
    project: `/api/projects/${props.id}`,
    updateProject: (data) => ({
        project: { 
          url: `/api/projects/${props.id}`, 
          method: "PATCH", 
          body: data,
          andThen: () => ({
            projectUpdated: { 
              value: true // some static value of your choice
            }
          })
        }
    })
}))
```

**Comment by georgebonnr** [comment]
Trying refetch out tonight, I was just about to ask about this, to help with transformations.  Any timeline, or is this a request for a PR?

**Comment by ryanbrainard** [comment]
@georgebonnr 

> I was just about to ask about this, to help with transformations.

Identity requests will allow for transformations of responses inside of `connect()`; however, you can use `PromiseState.then()` to do the same thing in the component. The only downsides of doing it that way are 1) gets applied for any prop change and 2) could clutter up the component if complex. Here's an example to transform the `usersFetch` to only include the verified users:

    this.props.usersFetch.then(users => users.filter(u => u.verified))

> Any timeline, or is this a request for a PR

I'd like to get to this soon, but we don't actually have a real need for it at this point with our own application of React Refetch, so that's slowing it down a bit. Also, there are a few different asks about transformations, handling different data types, etc, and I'd like to think about them holistically before rushing on any of them in piecemeal; however, I feel pretty confident about this one adding value with or without the others. I might even play around with it tonight since it should be fairly strait forward. With that said, anyone is welcome to open a PR for any of these enhancements.

**Comment by ryanbrainard** [comment]
Implemented in #36

**Comment by georgebonnr** [comment]
> Identity requests will allow for transformations of responses inside of connect(); however, you can use PromiseState.then() to do the same thing in the component.

Cool, yeah that's what I was planning on doing until identity is implemented – but no need now!  :clap:

**Comment by ryanbrainard** [comment]
Merged #36 and released in 0.7.0-beta.0. Please kick the tires :)

**Comment by hnordt** [comment]
Great @ryanbrainard. Simple and very efficient :clap:
